tests/test_de_example.py

```diff
@@ -0,0 +1,104 @@
+import unittest
+from collections import defaultdict
+
+from wikitextprocessor import Wtp
+
+from wiktextract.config import WiktionaryConfig
+from wiktextract.extractor.de.example import extract_examples, extract_reference
+
+from wiktextract.thesaurus import close_thesaurus_db
+from wiktextract.wxr_context import WiktextractContext
+
+
+class TestDEExample(unittest.TestCase):
+    maxDiff = None
+
+    def setUp(self) -> None:
+        self.wxr = WiktextractContext(
+            Wtp(lang_code="de"), WiktionaryConfig(dump_file_lang_code="de")
+        )
+
+    def tearDown(self) -> None:
+        self.wxr.wtp.close_db_conn()
+        close_thesaurus_db(
+            self.wxr.thesaurus_db_path, self.wxr.thesaurus_db_conn
+        )
+
+    def test_de_extract_examples(self):
+        self.wxr.wtp.start_page("")
```

Comment:
Please also add the Wiktionary word page links or titles used in tests comments.

Reply:
You mean links to the Wiktionary pages where I take example snippets from? Like this?: 
```
def test_de_extract_reference_from_literatur_template(self):
   # https://de.wiktionary.org/wiki/Beispiel```

You will see that some test cases I have are constructed. But I added references where appropriate and will try to use test cases from real Wiktionary pages going forward.

Reply:
Yes, but not required, just a recommendation for cases that are not easy to find the original page.